### ISSUE TITLE
UI freeze monitoring should be possible also in tests

### ISSUE THREAD

**Issue by frankbenoit** [eclipse-platform/eclipse.platform.ui#1670]
In an RCP application I want to use the UI freeze monitoring.

I found that the activation is depending on `Platform.inDevelopmentMode()`. 
This method detects if the application was started with the `-dev` option.
I assume this is to avoid events to be triggered if a debugger is connected or if the application is started from inside an IDE. 

However, also maven tycho surefire uses this option, hence in our Jenkins builds, UI freezes are not detectable.

Could you make it possible to have this detection available in maven builds as well?

### Proposal solution

Have a test method which detects if a debugger is present.
````
RuntimeMXBean runtime = ManagementFactory.getRuntimeMXBean();
return runtime.getInputArguments().toString().contains("jdwp");
````

- [x] I understand reporting an issue to this OSS project does not mandate anyone to fix it. Other contributors may consider the issue, or not, at their own convenience. The most efficient way to get it fixed is that I fix it myself and contribute it back as a good quality patch to the project.

**Comment by iloveeclipse** [comment]
We have freeze monitoring enabled in SDK, without starting it in debug mode. So why is this a problem to have it enabled in RCP?

**Comment by frankbenoit** [comment]
It is not active in Jenkins surefire, is it? Because this is what I am missing.
The reason I found is, that surefire makes use of `-dev` option, hence the freeze monitoring is disabled.

**Comment by iloveeclipse** [comment]
OK, this was misleading. So freeze monitoring **is** working in RCP but not working in tests. Would be easier to understand if the title would say that.

Contributions are welcome.

**Comment by laeubi** [comment]
> OK, this was misleading. So freeze monitoring **is** working in RCP but not working in tests. Would be easier to understand if the title would say that.

No it is not working if one passes the [-dev option](https://help.eclipse.org/latest/topic/org.eclipse.platform.doc.isv/reference/misc/runtime-options.html?cp=2_1_6_0) to RCP/launch/Eclipse/Whatever...

**Comment by frankbenoit** [comment]
@laeubi yes I did describe this, I know it is not working at the moment if the `-dev` option is passed. 

https://github.com/eclipse-platform/eclipse.platform.ui/blob/master/bundles/org.eclipse.ui.monitoring/src/org/eclipse/ui/internal/monitoring/MonitoringStartup.java#L46

I guess, this option cannot be avoided on surefire side.

Now I want to have the behavior changed, so the UI freeze events are also logged in surefire tests.
Perhaps by detecting a debugger, instead of using the `-dev` argument as the condition.

**Comment by laeubi** [comment]
@frankbenoit this was regarding @iloveeclipse comment that it does nor working in tests (what is wrong), it is just a sideeffect because the surefire-test are using the `-dev` option (as you correctly described).
